clients/src/main/java/org/apache/kafka/clients/consumer/internals/metrics/HeartbeatMetricsManager.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals.metrics;
+
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.Measurable;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Meter;
+import org.apache.kafka.common.metrics.stats.WindowedCount;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.kafka.clients.consumer.internals.ConsumerUtils.CONSUMER_METRIC_GROUP_PREFIX;
+import static org.apache.kafka.clients.consumer.internals.ConsumerUtils.COORDINATOR_METRICS_SUFFIX;
+
+public class HeartbeatMetricsManager {
```

Comment:
I was checking why the CI was misbehaving... it's probably not related to the PR, but while I was checking it, I noticed that other metrics containers like `KafkaConsumerMetrics` are `AutoClosable` and are explicitly closed inside `LegacyKafkaConsumer` and `AsyncKafkaConsumer`, removing all sensors from the `Metrics` class. Now, I'm not sure if this would cause any leaks, but it seems it would at least be good style to close metrics managers in the same way. What do you think? Sorry, didn't notice this on the first pass.

Reply:
I thought about this during the implementation.  However, the current RequestManager is not "closeable" nor do we try to close anything during shutdown.  I assume these sensors and metric map will be GC'ed upon program termination/fatal failure, so it occur to be a strange thing to do to remove these sensors from the map before closing.

If we need to - we should make RequestManager closeable as well as the metrics, so that we can close these resources upon program termination.

Reply:
GC should work fine with cyclic references. I suspect the point is that I can close my consumer, and I don't have to wait for GC to have the metrics cleaned up. You might even want to keep a reference to the consumer after it's closed, then GC will not kick in. 

I'm not entirely sure if this is just a "nice-to-have" or if there is a strong reason to do this, but since adding a couple of close methods should be a quick thing, I would vote for cleaning the sensors up explicitly.

Reply:
hey @lucasbru - The real issue is here: https://github.com/apache/kafka/pull/7839

The PR claims it is only affecting KafkaConsumerMetrics,.

Reply:
Okay, if we are not affected by this, not closing it is okay with me.